src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -0,0 +1,244 @@
+                    if (platformName != null)
+                    {
+                        info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, new Version(0, 0), negated: false);
+                        return true;
+                    }
+                }
+                else
+                {
+                    if (TryDecodeRuntimeInformationIsOSPlatform(arguments[0].Value, osPlatformType, out string? osPlatformName))
+                    {
+                        info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, osPlatformName, new Version(0, 0), negated: false);
+                        return true;
+                    }
+
+                    if (arguments[0].Value is ILiteralOperation literal &&
+                        literal.Type?.SpecialType == SpecialType.System_String &&
+                        literal.ConstantValue.HasValue)
+                    {
+                        // OperatingSystem.IsOSPlatform(string platform)
+                        if (invokedPlatformCheckMethod.Name == IsOSPlatform &&
+                            TryParsePlatformNameAndVersion(literal.ConstantValue.Value.ToString(), out string platformName, out Version? version))
+                        {
+                            info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, version, negated: false);
+                            return true;
+                        }
+
+                        // OperatingSystem.IsOSPlatformVersionAtLeast(string platform, int major, int minor = 0, int build = 0, int revision = 0)
+                        if (TryDecodeOSVersion(arguments, valueContentAnalysisResult, out version, 1))
+                        {
+                            info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, literal.ConstantValue.Value.ToString(), version, negated: false);
+                            return true;
+                        }
+                    }
+                    else if (arguments[0].Value is ILiteralOperation intLiteral && intLiteral.Type?.SpecialType == SpecialType.System_Int32)
+                    {
+                        // Accelerators like OperatingSystem.IsPlatformNameVersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)
+                        var platformName = SwitchVersionedPlatformName(invokedPlatformCheckMethod.Name);
+
+                        if (platformName != null && TryDecodeOSVersion(arguments, valueContentAnalysisResult, out var version))
+                        {
+                            info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, version, negated: false);
+                            return true;
+                        }
+                    }
+                }
+
+                info = default;
+                return false;
+            }
+
+            private static string? SwitchVersionedPlatformName(string methodName)
```

Comment:
I think we can avoid all the hard coding of platform names in this file  with simple regex based matches in both the methods added here. Code should be much more concise and we don’t need any changes in future if we add a new platform with corresponding IsPlatform and IsPlatformVersionAtLeast APIs.

Reply:
I was actually planning to update this use an immutable dictionary as it will be looked up very often, as you mentioned Regex might be better for future new methods but am a bit worried about performance. The analyzer was pretty slow with big source having 5k lines

Reply:
I agree with @mavasani except for the part where he said regex ;-)